server/routes/associatedPrisoner/associatePrisonerPage.ts

```diff
@@ -0,0 +1,237 @@
+/* eslint-disable max-classes-per-file */
+import { Request, Response } from 'express'
+import validateForm, { errors } from './associatePrisonerValidation'
+import { User } from '../../data/hmppsAuthClient'
+import PlaceOnReportService from '../../services/placeOnReportService'
+import adjudicationUrls from '../../utils/urlGenerator'
+import { DraftAdjudicationResult } from '../../data/DraftAdjudicationResult'
+import {
+  AssociatedPrisonerLocation,
+  getDraftIdFromString,
+  extractAssociatedDetails,
+  getPrisonerLocation,
+  RoleAssociatedPrisoner,
+  redirectToSearchForPersonPage,
+  redirectToDeletePersonPage,
+  renderData,
+  DisplayData,
+  redirectToOffenceSelection,
+  setRedirectUrl,
+  getAssociatedPrisonersNumber,
+  getAssociatedPrisonersName,
+} from './associatePrisonerUtils'
+import PrisonerSearchService from '../../services/prisonerSearchService'
+
+export enum PageRequestType {
+  EDIT,
+  EDIT_SUBMITTED,
+}
+
+class PageOptions {
+  constructor(private readonly pageType: PageRequestType) {}
+
+  isPreviouslySubmitted(): boolean {
+    return this.pageType === PageRequestType.EDIT_SUBMITTED
+  }
+}
+
+export default class AssociatePrisonerPage {
+  pageOptions: PageOptions
+
+  constructor(
+    pageType: PageRequestType,
+    private readonly placeOnReportService: PlaceOnReportService,
+    private readonly prisonerSearchService: PrisonerSearchService
+  ) {
+    this.pageOptions = new PageOptions(pageType)
+  }
+
+  view = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const draftId = getDraftIdFromString(req.params.adjudicationNumber)
+    const { roleCode } = req.params
+    const fromDelete = req.query.personDeleted as string
+    const fromSearch = JSON.stringify(req.query.selectedPerson)?.replace(/"/g, '')
+    let roleAssociatedPrisoner = await this.readFromApi(draftId, user as User)
+
+    if (fromSearch) {
+      roleAssociatedPrisoner = {
+        prisonerNumber: roleAssociatedPrisoner.prisonerNumber,
+        currentAssociatedPrisonerNumber: fromSearch,
+      }
+    } else if (fromDelete) {
+      roleAssociatedPrisoner = {
+        prisonerNumber: roleAssociatedPrisoner.prisonerNumber,
+      }
+    }
```

Comment:
Could we do this better by having a separate function with these if statements in, and then we don't need to use 'let' and reassign?